crates/libcgroups/src/common.rs

```diff
@@ -335,10 +339,17 @@ pub struct CgroupConfig {
     pub container_name: String,
 }
 
-pub fn create_cgroup_manager(
+pub fn create_cgroup_manager_with_root(
+    root_path: &Path,
     config: CgroupConfig,
 ) -> Result<AnyCgroupManager, CreateCgroupSetupError> {
-    let cgroup_setup = get_cgroup_setup().map_err(|err| match err {
+    let root = if root_path.is_empty() {
+        Path::new(DEFAULT_CGROUP_ROOT)
+    } else {
+        root_path
+    };
```

Comment:
How about using `Option`?
https://doc.rust-lang.org/std/option/enum.Option.html#method.unwrap_or_default

Reply:
good idea. I've changed it.